### PR TITLE
Feature/update kernel

### PR DESCRIPTION
**Pull request by SolovyovDmitry**

### CODE REVIEW

File: src/CompanyService.Business/Commands/Company/EditCompanyCommand.cs [hunk 1]
```diff
@@ -46,8 +45,7 @@ public class EditCompanyCommand : IEditCompanyCommand
 
     public async Task<OperationResultResponse<bool>> ExecuteAsync(Guid companyId, JsonPatchDocument<EditCompanyRequest> request)
     {
-      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveCompanyData)
-        && !await _accessValidator.HasRightsAsync(Rights.AddEditRemoveCompanies))
+      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveCompanyData, Rights.AddEditRemoveCompanies))
```
[the remark this answers is not in the document]
Comment (SolovyovDmitry) [reply src/CompanyService.Business/Commands/Company/EditCompanyCommand.cs:48]: we have && here

File: src/CompanyService.Data/CompanyRepository.cs [hunk 2]
```diff
@@ -94,7 +94,7 @@ public Task<bool> DoesExistAsync()
 
     public Task<bool> DoesNameExistAsync(string name)
     {
-      return _provider.Companies.AnyAsync(x => string.Equals(x.Name.ToLower(), name.ToLower()));
+      return _provider.Companies.AnyAsync(x => x.Name == name);
```
[the remark this answers is not in the document]
Comment (SolovyovDmitry) [reply src/CompanyService.Data/CompanyRepository.cs:97]: Linq doesn't care about register by default. It needs changes when you want it to care

File: src/CompanyService.Data/CompanyUserRepository.cs [hunk 3]
```diff
@@ -61,14 +60,6 @@ public Task<DbCompanyUser> GetAsync(Guid userId)
...
     }
 
-    public Task<List<DbCompanyUser>> GetAsync(List<Guid> userIds)
-    {
-      return _provider.CompaniesUsers
-        .Include(cu => cu.Company)
-        .Where(u => userIds.Contains(u.UserId) && u.IsActive)
-        .ToListAsync();
-    }
-
```
[the remark this answers is not in the document]
Comment (SolovyovDmitry) [reply src/CompanyService.Data/CompanyUserRepository.cs:71]: no references at all

File: src/CompanyService.Business/Commands/CompanyUser/CreateCompanyUserCommand.cs [hunk 4]
```diff
@@ -44,8 +44,7 @@ public class CreateCompanyUserCommand : ICreateCompanyUserCommand
 
     public async Task<OperationResultResponse<Guid?>> ExecuteAsync(CreateCompanyUserRequest request)
     {
-      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveCompanyData) 
-        && await _accessValidator.HasRightsAsync(Rights.AddEditRemoveCompanies))
+      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveCompanyData, Rights.AddEditRemoveCompanies))
```
[the remark this answers is not in the document]
Comment (SolovyovDmitry) [reply src/CompanyService.Business/Commands/CompanyUser/CreateCompanyUserCommand.cs:47]: &&

File: src/CompanyService.Data/CompanyRepository.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (vorobevamv) [reply src/CompanyService.Data/CompanyRepository.cs:97]: mmm.. interesting. OK)

File: src/CompanyService.Data/CompanyUserRepository.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (vorobevamv) [reply src/CompanyService.Data/CompanyUserRepository.cs:71]: OK

**Review by vorobevamv — APPROVED** [review]

File: src/CompanyService.Business/Commands/ContractSubject/CreateContractSubjectCommand.cs [hunk 5]
```diff
@@ -44,8 +44,7 @@ public class CreateContractSubjectCommand : ICreateContractSubjectCommand
 
     public async Task<OperationResultResponse<Guid?>> ExecuteAsync(CreateContractSubjectRequest request)
     {
-      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveCompanyData)
-        || !await _accessValidator.HasRightsAsync(Rights.AddEditRemoveCompanies))
+      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveCompanyData, Rights.AddEditRemoveCompanies))
```
Comment (vorobevamv) [inline src/CompanyService.Business/Commands/ContractSubject/CreateContractSubjectCommand.cs:47]: why both? not any?

File: src/CompanyService.Business/Commands/ContractSubject/EditContractSubjectCommand.cs [hunk 6]
```diff
@@ -43,8 +43,7 @@ public class EditContractSubjectCommand : IEditContractSubjectCommand
 
     public async Task<OperationResultResponse<bool>> ExecuteAsync(Guid contractSubjectId, JsonPatchDocument<EditContractSubjectRequest> request)
     {
-      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveCompanyData)
-        || !await _accessValidator.HasRightsAsync(Rights.AddEditRemoveCompanies))
+      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveCompanyData, Rights.AddEditRemoveCompanies))
```
Comment (vorobevamv) [inline src/CompanyService.Business/Commands/ContractSubject/EditContractSubjectCommand.cs:46]: and here: both? not any?

File: src/CompanyService.Business/Commands/ContractSubject/CreateContractSubjectCommand.cs (hunk 5 above)
Comment (SolovyovDmitry) [reply src/CompanyService.Business/Commands/ContractSubject/CreateContractSubjectCommand.cs:47]: Old logic: if one of rights is absent return false so it means that we need both

Comment (vorobevamv) [reply src/CompanyService.Business/Commands/ContractSubject/CreateContractSubjectCommand.cs]: stay only CompanyData here

File: src/CompanyService.Data/CompanyUserRepository.cs [hunk 7]
```diff
@@ -108,7 +99,7 @@ public Task<bool> DoesExistAsync(Guid userId)
 
     public async Task<bool> RemoveContractSubjectAsync(Guid contractSubjectId)
     {
-      List<DbCompanyUser> dbUsers = await _provider.CompaniesUsers.Where(x => x.ContractSubjectId == contractSubjectId).ToListAsync();
+      IQueryable<DbCompanyUser> dbUsers = _provider.CompaniesUsers.Where(x => x.ContractSubjectId == contractSubjectId);
```
Comment (ndoriomedov) [inline src/CompanyService.Data/CompanyUserRepository.cs:102]: why u removed .ToListAsync()?

Comment (ndoriomedov) [reply src/CompanyService.Data/CompanyUserRepository.cs:102]: before iterating in foreach data anyway downloaded from db, but is it async operation?

Comment (SolovyovDmitry) [reply src/CompanyService.Data/CompanyUserRepository.cs:102]: > before iterating in foreach data anyway downloaded from db, but is it async operation?

I'll check this and revert if it is

Comment (SolovyovDmitry) [reply src/CompanyService.Data/CompanyUserRepository.cs:102]: seems to be right

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**